### PR TITLE
[GOBBLIN-1844] Ignore workflows marked for deletion when calculating container count

### PR DESCRIPTION
**Pull request by homatthew**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-1844] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1844

### Description
- [X] Here are some details about my PR, including screenshots (if applicable):
When the `GobblinClusterManager` starts, it will do an asynchronous deletion of workflows (all workflow deletions are technically async). And we do not want the yarn service to acknowledge these workflows while the controller processes the request asynchronously. 

When there is heavy load on the zookeeper cluster, the deletes will be slow and there will be significant overlap of having 2 of the same workflows 

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Modified existing unit to cover edge case of workflows that marked for deletion. Also refactored the testing code to improve the readability since there is a ton of copy paste involved for creating a workflow which is difficult to read

image

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-yarn/src/test/java/org/apache/gobblin/yarn/YarnAutoScalingManagerTest.java [hunk 1]
```diff
@@ -223,117 +150,64 @@ public void testTwoWorkflows() throws IOException {
...
 
   /**
-   * Test for two workflows with one not in progress.
-   * The partitions for the workflow that is not in progress should not be counted.
+   * Test for three workflows with one not in progress and one marked for delete.
+   * The partitions for the workflow that is not in progress or is marked for delete should not be counted.
    */
   @Test
-  public void testNotInProgress() throws IOException {
+  public void testNotInProgressOrBeingDeleted()  {
```
Comment (homatthew) [inline gobblin-yarn/src/test/java/org/apache/gobblin/yarn/YarnAutoScalingManagerTest.java:162]: This is the test that covers the related `impl` changes. The changes in the other test methods are all refactors

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnAutoScalingManager.java [hunk 2]
```diff
@@ -220,16 +220,18 @@ void runInternal() {
       YarnContainerRequestBundle yarnContainerRequestBundle = new YarnContainerRequestBundle();
       for (Map.Entry<String, WorkflowConfig> workFlowEntry : taskDriver.getWorkflows().entrySet()) {
         WorkflowContext workflowContext = taskDriver.getWorkflowContext(workFlowEntry.getKey());
+        WorkflowConfig workflowConfig = workFlowEntry.getValue();
 
-        // Only allocate for active workflows
-        if (workflowContext == null || !workflowContext.getWorkflowState().equals(TaskState.IN_PROGRESS)) {
+        // Only allocate for active workflows and those not marked for delete
```
Comment (ZihanLi58) [inline gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnAutoScalingManager.java:225]: Can you add a comment here to explain that we won't release the container immediately as we allow containers to be idle for a specific time?

**Review by ZihanLi58 — COMMENTED** [review]
LGTM overall, just a minor comment

**Review by ZihanLi58 — APPROVED** [review]
+1, thanks for addressing the comments
